src/messaging/ExchangeContext.cpp

```diff
@@ -537,6 +537,13 @@ CHIP_ERROR ExchangeContext::HandleMessage(uint32_t messageCounter, const Payload
     // layer has completed its work on the ExchangeContext.
     ExchangeHandle ref(*this);
 
+    // If we receive a message while we are expecting a response for our previous message,
+    // we assume that the peer is likely active. If the message was in fact the response,
+    // the flag will be reset to false when we process the message later in the function.
```

Comment:
```suggestion
    // If we receive a message while we are expecting a response for our previous message,
    // we assume that the peer is likely active, since it has not sent us that response yet and hence
    // still has an active exchange corresponding to this one.  If this message is in fact the response,
    // the flag will be reset to false when we process the message later in the function.
```

Reply:
The response that we received could also have requested an Ack and the peer would wait for that Ack and keep retransmitting the response until it receives it. So, here we are implicitly discounting that extra bit of time when we send the Ack and the peer waits for it. There is no good way to quantify that time since the Ack that we sent may have been received or it might get lost and the Peer actually stays active to retransmit. But as an estimate, we deem the Peer to no longer be active after the message response has been received on this exchange.
It would be good to capture this in the documentation also.

Reply:
> we deem the Peer to no longer be active after the message response has been received on this exchange.
It would be good to capture the above in the documentation also.

This is documented when we remove the hint later in the function when we know the message received is the expected response.